frontend/services/sources.go

```diff
@@ -319,29 +320,53 @@ func updateAnnotations(annotations map[string]string, reportedName string) map[s
 	return annotations
 }
 
-func GetSourceCRD(ctx context.Context, nsName string, workloadName string, workloadKind WorkloadKind) (*v1alpha1.Source, error) {
-	if workloadKind != WorkloadKindDeployment && workloadKind != WorkloadKindStatefulSet && workloadKind != WorkloadKindDaemonSet {
-		return nil, errors.New("unsupported workload kind " + string(workloadKind))
+func GetSourceCRDs(ctx context.Context, args ...interface{}) ([]*v1alpha1.Source, error) {
+	var nsName, workloadName string
+	var workloadKind WorkloadKind
+	if len(args) > 0 {
+		nsName, _ = args[0].(string)
+	}
+	if len(args) > 1 {
+		workloadName, _ = args[1].(string)
+	}
+	if len(args) > 2 {
+		workloadKind, _ = args[2].(WorkloadKind)
+		if workloadKind != WorkloadKindDeployment && workloadKind != WorkloadKindStatefulSet && workloadKind != WorkloadKindDaemonSet {
+			return nil, errors.New("unsupported workload kind " + string(workloadKind))
+		}
```

Comment:
im not sure why you desiced to pass those arg in this way and then re-aissned them to new vars

Reply:
because they are optional, you can either get the whole list with no args, or get the 1 source with args